lib/src/characters.dart

```diff
@@ -42,8 +30,13 @@ abstract class Characters implements Iterable<String> {
   /// A specialized character iterator.
   ///
   /// Allows iterating the characters of [string] as a plain iterator,
-  // as well as controlling the iteration in more detail.
-  Character get iterator;
+  /// as well as controlling the iteration in more detail.
+  CharacterRange get iterator;
+
+  /// A specialized character iterator positioned at the end.
+  ///
+  /// Allows iterating the characters of [string] backwards from the end.
+  CharacterRange get iteratorEnd;
```

Comment:
It would not be consistent with `List` because that's a one-directional iterator.
This is a bi-directional iterator currently at the end of the string. You still need to use `movePrevious` to go back from there.
(So: Documentation needs to be better).